### ISSUE TITLE
Flexbox calendar component broken in Firefox

### ISSUE THREAD

**Issue by TravellerSam** [zellwk/zellwk.com#37]
Your otherwise great flex/cssgrid combined [calendar widget](https://codepen.io/zellwk/pen/ZNrezV/) leaves the last column (Saturday) empty when the flexbox version is used on Firefox (68/macOS). The css grid version works correctly.

![Screenshot 2019-07-30 at 11 00 37](https://user-images.githubusercontent.com/872730/62116270-2cfdc300-b2ba-11e9-90dc-87da04d4ff55.png)

**Comment by zellwk** [comment]
Strange. I tested with Firefox 64 and it's perfectly ok. 
Tried upgrading to Firefox 68, and Firefox simply wouldn't load any page (not even google). So unfortunately, I can't debug this right now.

**Comment by TravellerSam** [comment]
Try getting the [Firefox Developer Edition](https://www.mozilla.org/en-US/firefox/developer/), which can run alongside your normal Firefox installation. Plus, it's great for debugging.

**Comment by zellwk** [comment]
I'm using Nightly, and it works fine.

**Comment by zellwk** [comment]
Closing this until there's more information about the problem.

**Comment by TravellerSam** [comment]
I'm not sure what other information you need. Your codepen example is broken in Firefox in the flexbox mode, as tested in stable and dev Firefox on Linux, macOS and Windows (7 & 10), and Nightly on macOS and Win10.

Your code works nicely in Chrome/Chromium and even in MS Edge on Win10, albeit with a slight layout issue (as in doesn't understand `max-width: max-content` for `main`).

**Comment by zellwk** [comment]
No idea why, I swear it works on my side. Requesting for more people to help test: https://twitter.com/zellwk/status/1160726626046337024?s=20

**Comment by huijing** [comment]
This appears to be a sub-pixel calculation issue. I inspected the original `4.5ch` computed values and found the width of the `date-grid` to be `350.183px`, but its `button` children have a computed width of `50.0333px`. When you multiply that value by 7, it ends up being `350.2331px` which is larger than your flex container, hence it drops to the next line instead of filling the column.

When I changed the value to `4ch` instead, the calculation works out to be a flex container width of `311.267px`, and each flex child being `44.4667px`, which when multiplied by 7, gives you `311.2669px` hence still falls within the flex container width and resolves the empty column issue.
